core/services/pipeline/models.go

```diff
@@ -0,0 +1,63 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/guregu/null.v4"
+)
+
+type (
+	Spec struct {
+		ID           int32 `gorm:"primary_key"`
+		DotDagSource string
```

Comment:
I remember seeing something very similar to this on another model somewhere...

Reply:
Why is it stored twice?

Reply:
It's stored on the `pipeline.TaskDAG`, which is not persisted to the DB.  The TaskDAG, a type that only exists for the purposes of unmarshaling DOT into Go structs, simply hands it off to the `pipeline.Spec` (which _is_ persisted).